cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -0,0 +1,158 @@
+"""Management command for analyzing AWS CloutTrail Logs."""
+import json
+
+from django.core.management.base import BaseCommand
+from django.utils.translation import gettext as _
+
+# from account.models import Account, Instance, InstanceEvent
+from util import aws
+
+
+class Command(BaseCommand):
+    """Command to analyze EC2 events from AWS CloudTrail logs.
+
+    Example usage:
+
+        QUEUE_URL="https://sqs.us-east-1.amazonaws.com/372779871274/doppler_customer_1_test_bucket_cloudtrail_events"
+        python manage.py analyze_log $QUEUE_URL
+    """
+
+    help = _('Analyzed AWS CloudTrail logs for EC2 on/off events.')
+
+    def add_arguments(self, parser):
+        """Require an AWS SQS queue URL."""
+        parser.add_argument('queue_url', help=_('AWS SQS Queue URL'))
+
+    def handle(self, *args, **options):
+        """Read SQS Queue for log location, and parse log for events."""
+        queue_url = options['queue_url']
+
+        logs = []
+        parsed_messages = []
+
+        # Get messages off of an SQS queue
+        messages = aws.receive_message_from_queue(queue_url)
+
+        # Parse the SQS messages to get S3 object locations
+        for message in messages:
+            parsed_messages.extend(self._parse_sqs_message(message))
+
+        # Grab the object contents from S3
+        for parsed_message in parsed_messages:
+            bucket = parsed_message[0]
+            key = parsed_message[1]
+            logs.append(aws.get_object_content_from_s3(bucket, key))
+
+        # Parse logs for on/off events
+        for log in logs:
+            if log is None or log == '':
+                continue
+            instances, instance_events = self._parse_log_for_ec2_events(log)
+            result = 'Found instances: {i} and events {e}'.format(
+                i=instances, e=instance_events)
+            self.stdout.write(_(result))
+
+        # TODO: Save the messages to the DB
+        # TODO: If we haven't seen this instance before we'll
+        #       need to run describe_instances on it.
+        #       Start and stop requests don't have all the
+        #       info we need.
+
+        # aws.delete_message_from_queue(queue_url, messages)
+
+    def _parse_sqs_message(self, message):
```

Comment:
Part of me feels like this belongs in the `aws` module as a more general-purpose function next to `receive_message_from_queue`, but I'm OK with leaving it here until something else needs to read from the queue.

Reply:
I can be swayed. I put it here because there can be different message types in SQS. This function isn't, in its current form, generalized enough to accomodate that, but it *IS* a util for what we want now, so I'll gladly put it there now. If we ever see another kind of SQS message, we can fix it then.